cpp/tests/reductions/reduction_tests.cu

```diff
@@ -756,4 +755,125 @@ TEST_F(StringReductionTest, AllNull)
+
+  this->reduction_test(col_nulls, expected_null_value, this->ret_non_arithmetic,
+                       cudf::experimental::make_median_aggregation());
+}
+
+TYPED_TEST(ReductionTest, Quantile)
+{
+  using T = TypeParam;
+  //{-20, -14, -13,  0, 6, 13, 45, 64/None} 
+  std::vector<int> int_values({6, -14, 13, 64, 0, -13, -20, 45});
+  std::vector<bool> host_bools({1, 1, 1, 0, 1, 1, 1, 1});
+  std::vector<T> v = convert_values<T>(int_values);
+  cudf::experimental::interpolation interp{cudf::experimental::interpolation::LINEAR};
+
+  // test without nulls
+  cudf::test::fixed_width_column_wrapper<T> col(v.begin(), v.end());
+  double expected_value0 = std::is_same<T, cudf::experimental::bool8>::value ? v[4] : v[6];
+  this->reduction_test(col, expected_value0, this->ret_non_arithmetic,
+                       cudf::experimental::make_quantile_aggregation({0.0}, interp));
+  double expected_value1 = v[3];
+  this->reduction_test(col, expected_value1, this->ret_non_arithmetic,
+                       cudf::experimental::make_quantile_aggregation({1.0}, interp));
+
+  // test with nulls
+  cudf::test::fixed_width_column_wrapper<T> col_nulls = construct_null_column(v, host_bools);
+  double expected_null_value1 = v[7];
+
+  this->reduction_test(col_nulls, expected_value0, this->ret_non_arithmetic,
+                       cudf::experimental::make_quantile_aggregation({0}, interp));
+  this->reduction_test(col_nulls, expected_null_value1, this->ret_non_arithmetic,
+                       cudf::experimental::make_quantile_aggregation({1}, interp));
+}
+
+TYPED_TEST(ReductionTest, UniqueCount)
+{
+  using T = TypeParam;
+  std::vector<int>  int_values({1,-1, 1, 2, 0, 2,-2, 45}); //6
+  std::vector<bool> host_bools({1, 1, 1, 0, 1, 1, 1, 1});  //6
+  std::vector<T> v = convert_values<T>(int_values);
+
+  // test without nulls
+  cudf::test::fixed_width_column_wrapper<T> col(v.begin(), v.end());
+  cudf::size_type expected_value = std::is_same<T, cudf::experimental::bool8>::value ? 2 : 6;
+  this->reduction_test(col, expected_value, this->ret_non_arithmetic, cudf::experimental::make_nunique_aggregation(cudf::include_nulls::YES));
+  this->reduction_test(col, expected_value, this->ret_non_arithmetic, cudf::experimental::make_nunique_aggregation(cudf::include_nulls::NO));
+
+  // test with nulls
+  cudf::test::fixed_width_column_wrapper<T> col_nulls = construct_null_column(v, host_bools);
+  cudf::size_type expected_null_value0 = std::is_same<T, cudf::experimental::bool8>::value ? 3 : 7;
+  cudf::size_type expected_null_value1 = std::is_same<T, cudf::experimental::bool8>::value ? 2 : 6;
+
+  this->reduction_test(col_nulls, expected_null_value0, this->ret_non_arithmetic, cudf::experimental::make_nunique_aggregation(cudf::include_nulls::YES));
+  this->reduction_test(col_nulls, expected_null_value1, this->ret_non_arithmetic, cudf::experimental::make_nunique_aggregation(cudf::include_nulls::NO));
+}
+
+
+TYPED_TEST(ReductionTest, NthElement)
```

Comment:
Negative tests are missing (an in-out combination that falls under `is_not_supported()`